### PR TITLE
Initial version of the Supplychain transaction processor.

### PR DESCRIPTION
**Pull request by cintel**
Signed-off-by: Cian <cian.montgomery@intel.com>

### CODE REVIEW

**Comment by cintel** [comment]
On this Vagrant instance I am getting the bogus lint errors, the python path is correctly set to '/project/sawtooth-core/sdk/python:/project/sawtooth-core/sdk/examples/intkey_python:/project/sawtooth-core/sdk/examples/supplychain_python:/project/sawtooth-core/sdk/examples/xo_python:/project/sawtooth-core/signing' in run_lint prior to the lint invocation.

************* Module processor.handler
E: 24, 0: Unable to import 'sawtooth_supplychain.addressing' (import-error)
************* Module processor.main
E: 25, 0: Unable to import 'sawtooth_supplychain.processor.handler' (import-error)
************* Module supplychain_client
E: 29, 0: Unable to import 'sawtooth_supplychain.supplychain_exceptions' (import-error)
E: 31, 0: Unable to import 'sawtooth_supplychain.addressing' (import-error)
E: 32, 0: Unable to import 'sawtooth_supplychain.time_utils' (import-error)
************* Module supplychain_message_factory
E: 19, 0: Unable to import 'sawtooth_supplychain.processor.handler' (import-error)
************* Module test_tp_supplychain
E: 25, 0: Unable to import 'sawtooth_supplychain.supplychain_message_factory' (import-error)
E: 27, 0: Unable to import 'sawtooth_supplychain.addressing' (import-error)

File: bin/tp_intkey_python [hunk 1]
```diff
@@ -25,15 +25,15 @@ build_str = "lib.{}-{}.{}".format(
 
 sys.path.insert(0, os.path.join(
         os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
-            'sdk', 'examples', 'intkey_python'))
+            'sdk', 'examples', 'supplychain_python'))
```
Comment (peterschwarz) [inline bin/tp_intkey_python]: Looks like a bad find-and-replace

File: bin/tp_intkey_python [hunk 2]
```diff
@@ -25,15 +25,15 @@ build_str = "lib.{}-{}.{}".format(
...
 sys.path.insert(0, os.path.join(
         os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
             'sdk', 'python'))
 sys.path.insert(0, os.path.join(
         os.path.dirname(os.path.dirname(os.path.realpath(__file__))),
             'signing'))
 
-from sawtooth_intkey.processor.main import main
+from sawtooth_supplychain.processor.main import main
```
Comment (peterschwarz) [inline bin/tp_intkey_python]: same as above

File: sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import json
+
+from sawtooth_processor_test.message_factory import MessageFactory
+from sawtooth_supplychain.processor.handler import SUPPLYCHAIN_NAMESPACE
+
+
+class SupplychainMessageFactory(object):
```
Comment (peterschwarz) [inline sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py]: Since this is test code, this probably shouldn't live under the production code tree

Comment (cintel) [inline sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py]: Was following the pattern of intkey_python, Is there a better pattern show in the repo to follow?

**Comment by vaporos** [comment]
If this is less of an SDK example, it could also go in families/supplychain/.  Other families there are families/config/ (just moved) and families/burrow_evm/ (in the feature branch currently).

File: sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py (hunk 3 above)
Comment (peterschwarz) [inline sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py]: The pattern was incorrect, and I've started to move these classes for their respective transaction processors (config was first).

Comment (cintel) [inline sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py]: @peterschwarz is there a pr I can reference for the new desired pattern?

**Comment by cintel** [comment]
I am coordinating the Supplychain TP spec with Sean and simon, the original authors. Will add a PR for it when it is publically consumable. I am not sure it is worth holding this PR untill it arrives.

File: sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py (hunk 3 above)
Comment (peterschwarz) [inline sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py]: Commit https://github.com/hyperledger/sawtooth-core/pull/495/commits/ba50ebc3dee8c482d52f9683b5297c66b8a7e2c7 from PR #495

Comment (peterschwarz) [inline sdk/examples/supplychain_python/sawtooth_supplychain/supplychain_message_factory.py]: You could do it as a separate PR, but since the code is new seems like a good time to just fix it.

**Comment by cintel** [comment]
@peterschwarz updated location as per the referenced PR.
